triples-generator/src/test/scala/io/renku/triplesgenerator/events/consumers/tsmigrationrequest/migrations/MigrationToNamedGraphsSpec.scala

```diff
@@ -0,0 +1,203 @@
+
+      val defaultOnlyProjects = projectInfos.generateList(min = 20, max = 30)
+      val bothDSProjects      = projectInfos.generateNonEmptyList().toList
+
+      val allDefaultDSProjectsChunks = (defaultOnlyProjects ::: bothDSProjects)
+        .sortBy(_._2)
+        .sliding(chunkSize, chunkSize)
+        .toList
+      allDefaultDSProjectsChunks.zipWithIndex foreach givenProjectsChunkReturned
+      givenProjectsChunkReturned(List.empty[ProjectInfo] -> allDefaultDSProjectsChunks.size)
+
+      defaultOnlyProjects foreach (givenProject(_, existsInNamedGraphs = false))
+      bothDSProjects foreach (givenProject(_, existsInNamedGraphs = true))
+
+      defaultOnlyProjects map toRedoEvent foreach givenEventIsSent
+
+      migration.run().value.unsafeRunSync() shouldBe ().asRight
+    }
+
+    "return a Recoverable Error if in case of an exception while finding projects " +
+      "the given strategy returns one" in new TestCase {
+
+        val exception = exceptions.generateOne
+
+        (defaultGraphProjectsFinder.findDefaultGraphProjects _)
+          .expects(1)
+          .returning(exception.raiseError[IO, List[ProjectInfo]])
+
+        migration.run().value.unsafeRunSync() shouldBe recoverableError.asLeft
+      }
+  }
+
+  private trait TestCase {
+    val chunkSize: Int = 30
+
+    val eventCategoryName          = CategoryName("EVENTS_STATUS_CHANGE")
+    val defaultGraphProjectsFinder = mock[DefaultGraphProjectsFinder[IO]]
+    val namedGraphsProjectFinder   = mock[NamedGraphsProjectFinder[IO]]
+    val eventSender                = mock[EventSender[IO]]
+    val recoverableError           = processingRecoverableErrors.generateOne
+    val recoveryStrategy = new RecoverableErrorsRecovery {
+      override def maybeRecoverableError[F[_]: MonadThrow, OUT]: RecoveryStrategy[F, OUT] = { _ =>
+        recoverableError.asLeft[OUT].pure[F]
+      }
+    }
+    val migration = new MigrationToNamedGraphs[IO](defaultGraphProjectsFinder,
+                                                   namedGraphsProjectFinder,
+                                                   eventSender,
+                                                   recoveryStrategy
+    )
+
+    def givenProjectsChunkReturned: ((List[ProjectInfo], Int)) => Unit = { case (chunk, idx) =>
+      (defaultGraphProjectsFinder.findDefaultGraphProjects _)
+        .expects(idx + 1)
+        .returning(chunk.pure[IO])
+      ()
+    }
+
+    def givenProject(project: ProjectInfo, existsInNamedGraphs: Boolean): Unit =
+      givenProject(project, existsInNamedGraphs.pure[IO])
+
+    def givenProject(project: ProjectInfo, existsInNamedGraphsResponse: IO[Boolean]): Unit = {
+      (namedGraphsProjectFinder.checkExists _)
+        .expects(project._1)
+        .returning(existsInNamedGraphsResponse)
+      ()
+    }
+
+    def toRedoEvent(project: ProjectInfo) = project -> EventRequestContent.NoPayload {
+      json"""{
```

Comment:
Wouldn't it be beneficial if we would have a case class representing the payload of these events instead of writing the literal json each time?

Reply:
Right, that's a bit more complicated thing as it goes beyond a single service. I mean, we can think if we'd like to introduce some special module that lists event payload classes or do it in some other way as I think it's rather not the best idea to simply make EL a compile-time dependency of TG.
Let's maybe leave it for now and discuss maybe at some other time.

Reply:
sure, leave it for now - I was wondering more in general, unrelated to this pr. 👍🏼